tests/functional/build-remote.sh

```diff
@@ -70,6 +71,12 @@ echo "$output" | grepQuietInverse builder-build-remote-input-2.sh
 echo "$output" | grepQuiet builder-build-remote-input-3.sh
 unset output
 
+# Ensure that input4 was built on store4 due to the required feature.
```

Comment:
This isn't obvious from the code below. Could you add assertions that simply check what you're saying here?

Reply:
I tried getting a test here which would reject builds in the proper way but I'm not confident with the testing framework so I might need guidance.